ui/src/reducers/search.js

```diff
@@ -5,104 +5,203 @@ import {
   SEARCH_REQUEST,
   SEARCH_ERROR,
   SEARCH_SUCCESS,
-  CHANGE_SEARCH_SCOPE,
+  CHANGE_SEARCH_BOX_NAMESPACE,
   SEARCH_AGGREGATIONS_REQUEST,
   SEARCH_AGGREGATIONS_SUCCESS,
   SEARCH_AGGREGATIONS_ERROR,
   NEW_SEARCH_REQUEST,
+  SEARCH_QUERY_UPDATE,
+  SEARCH_BASE_QUERIES_UPDATE,
 } from '../actions/actionTypes';
-import { AUTHORS, JOBS } from '../common/routes';
+import { AUTHORS, JOBS, LITERATURE } from '../common/routes';
 
-const baseQuery = {
+export const AUTHORS_NS = 'authors';
+export const LITERATURE_NS = 'literature';
+export const JOBS_NS = 'jobs';
+export const AUTHOR_PUBLICATIONS_NS = 'authorPublications';
+
+const initialBaseQuery = {
   sort: 'mostrecent',
   size: '25',
   page: '1',
 };
 
-export const searchScopes = fromJS({
-  literature: {
-    name: 'literature',
-    pathname: 'literature',
-    query: baseQuery,
-  },
-  authors: {
-    name: 'authors',
-    pathname: 'authors',
-    query: baseQuery,
-  },
-  jobs: {
-    name: 'jobs',
-    pathname: 'jobs',
-    query: baseQuery,
-  },
-});
-
-export const initialState = fromJS({
+const initialNamespaceState = {
   loading: false,
   total: 0,
-  scope: searchScopes.get('literature'),
   error: null,
   sortOptions: null,
   aggregations: {},
   initialAggregations: {},
   loadingAggregations: false,
   aggregationsError: null,
+  baseQuery: initialBaseQuery,
+  query: initialBaseQuery,
+  baseAggregationsQuery: {},
+};
+
+export const FETCH_MODE_NEVER = 'never';
+export const FETCH_MODE_ALWAYS = 'always';
+export const FETCH_MODE_INITIAL = 'only-initial-without-query';
+
+export const initialState = fromJS({
+  searchBoxNamespace: LITERATURE_NS,
+  namespaces: {
+    [LITERATURE_NS]: {
+      ...initialNamespaceState,
+      pathname: LITERATURE,
+      embedded: false,
```

Comment:
maybe embedded could go as false by default?

Reply:
I want it to be explicitly state for every namespace.